tests/test_dependencies.py

```diff
@@ -326,6 +328,56 @@ def test_load_save_backward_compatibility(tmpdir, deps):
     assert deps == deps2
 
 
+@pytest.mark.parametrize("pandas_version", ["2.0.3", "2.1.4", "2.2.2"])
+def test_load_save_pandas_compatibility(pandas_version):
+    """Test pandas backward compatibility of pickle cache files.
+
+    Dataframes using pyarrow dtypes,
+    and stored as pickle files
+    might fail to load
+    if the used ``pandas`` version
+    does not match.
+
+    This tests documents,
+    which cases are problematic,
+    and need to be handled inside
+    ``audb.dependencies()``.
+
+    See Also:
+    https://github.com/audeering/audb/issues/418
+
+    Args:
+        pandas_version: the version of ``pandas``
+            used to store the dependency table in cache
+
+    """
+    deps_file = audeer.path(
+        CURRENT_DIR,
+        "assests",
+        "dependency-table-pandas",
+        f"emodb-pandas-{pandas_version}.pkl",
+    )
+    deps = audb.Dependencies()
+
+    # Dependency table cached with pandas==2.0.3.
+    # Loading with pandas>=2.1.0 leads to a ModuleNotFoundError
+    if pd.__version__ >= "2.1.0" and pandas_version == "2.0.3":
```

Comment:
I am not 100% happy with the conditionals here. I wonder whether this could be handled in the `@pytest.mark.parametrize` decorator? Something like [this](https://docs.pytest.org/en/7.1.x/example/parametrize.html#parametrizing-conditional-raising:~:text=Parametrizing%20conditional%20raising,is%20not%20None) would be more idiomatic, wouldn't it?  
I will not insist on this in the review, :-)

Reply:
I'm also not happy with it.

I tried an approach using a fixture in https://github.com/audeering/audb/pull/428,
but when running the test I get already an error as `pytest-virtualenv` is not in line with our `pyproject.toml` file, but expects a `setup.py` file:

```
/tmp/tmpewep4i73/.env/bin/python setup.py -q develop' returned non-zero exit
```

Reply:
In addition, the definition of the test is not really easier to understand now.
So maybe we just stay with the current solution?